llvm/lib/Transforms/Vectorize/LoopVectorizationLegality.cpp

```diff
@@ -1601,8 +1601,24 @@ bool LoopVectorizationLegality::prepareToFoldTailByMasking() {
 
   LLVM_DEBUG(dbgs() << "LV: can fold tail by masking.\n");
 
-  MaskedOp.insert(TmpMaskedOp.begin(), TmpMaskedOp.end());
   return true;
 }
 
+void LoopVectorizationLegality::prepareToFoldTailByMasking() {
```

Comment:
Above comments worth updating:
```
  // Collect masked ops in temporary set first to avoid partially populating
  // MaskedOp if a block cannot be predicated.
```
comment needs rephrasing.

```
  // Check and mark all blocks for predication, including those that ordinarily
  // do not need predication such as the header block.
```
only Check, w/o mark

```
LLVM_DEBUG(dbgs() << "LV: Cannot fold tail by masking as requested.\n");
```
drop "as requested"?

Reply:
Updated, thanks!